Small home page UI improvements

florian-lefebvre opened the pull request:
## What kind of change does this PR introduce?

Home page UI improvements

florian-lefebvre committed:
small changes

florian-lefebvre committed:
fetch upstream

florian-lefebvre committed:
small ui improvements

florian-lefebvre committed:
update dev doc for Windows users

florian-lefebvre committed:
remove yarn.lock

kiwicopple commented:
Thanks for this @florian-lefebvre. Would you mind dropping in a quick screenshot of the changes?

Great update on `development.md`

florian-lefebvre committed:
revert changes on FlyOut component and update Theme function instead

florian-lefebvre committed:
fixes

florian-lefebvre commented:
No screenshots to show @kiwicopple  because no major changes but only small improvements:
- There was an issue with the FlyOut component, it was larger than the page so I put `overflow-x-hidden` on the html element and had to modify the Theme function
- There was the wrong focus color (indigo instead of brand) on one button (last one)
- I changed a bit some colors of the mobile menu because I thought there wasn't enough contrast
- I fixed some colors not adapted to dark mode
- There was an issue with the mobile menu: when opening it, increasing the screen width hid this menu but overflow was hidden on the body so I added `lg:overflow-auto`

sw-yx commented on www/package.json:
```diff
@@ -46,9 +46,10 @@
     "postcss-preset-env": "^6.7.0",
     "prettier": "^2.2.1",
-    "tailwindcss": "^2.0.1"
+    "tailwindcss": "2.0.4"
```
just curious why you had to pin versions?

florian-lefebvre replied:
I just did it by reflex. But I added PostCSS 8 because otherwise I couldn't run next.

MildTomato commented on www/pages/_document.tsx:
```diff
@@ -3,9 +3,9 @@ import Document, { Html, Head, Main, NextScript } from 'next/document'
+      <Html lang="en" className="dark overflow-x-hidden">
         <Head></Head>
-        <body>
+        <body className="lg:overflow-auto">
```
what are these changes for?

florian-lefebvre replied:
For now, on pc, if you reduce the screen width (under the lg breakpoint), you will be in mobile configuration. You can then open the mobile menu, which disables scrolling on the body. But if you enlarge the width of the screen, the mobile menu is hidden but you still can't scroll the body. So with `lg:overflow-auto` we ensure the body is always scrollable on larger screen sizes.

MildTomato replied:
Gotcha! makes sense now 😄 

I think a better way to deal with this could be just to make the mobile menu full width, and remove the hidden property for lg screens on it.

something like the below might mean you don't have to put in utility css classes on`<body>`.

```js
// supabase/www/components/Nav/index.tsx

// line 285
 <div className="p-4 md:p-8 h-screen w-screen fixed bg-white transform overflow-y-scroll -inset-y-0 z-50 dark:bg-dark-700"> ```

florian-lefebvre replied:
Yes it's better like this, I changed it.

MildTomato commented on development.md:
```diff
@@ -6,6 +6,10 @@ The beta website has a `basePath` of `new` and using `vercel.json`. This is to d
 
 To run this locally, run the following commands. This will start the website at http://localhost:8080/ and requests will be automatically proxied (via vercel rewrites) to the docs website (running at http://localhost:3005/) and the beta website (running at http://localhost:3000)
 
+On Windows, you will need to install [Make for Windows](http://gnuwin32.sourceforge.net/packages/make.htm) and then to add `C:\Program Files (x86)\GnuWin32\bin` to the system PATH **OR** just run the dev command in the `Makefile` directly in the terminal: `vercel dev --listen 8080 --local-config vercel-local.json`.
```
@florian-lefebvre nice work 👍

florian-lefebvre committed:
make mobile menu full width and remove body class

florian-lefebvre commented:
Any other question concerning my pull request ?

florian-lefebvre commented on www/pages/_document.tsx:
```diff
@@ -3,7 +3,7 @@ import Document, { Html, Head, Main, NextScript } from 'next/document'
   render() {
     return (
-      <Html lang="en" className="dark">
+      <Html lang="en" className="dark overflow-x-hidden">
```
No because it's linked to FlyOut width issue.
![image](https://user-images.githubusercontent.com/69633530/112822230-f7f5d500-9087-11eb-8668-c6e1014a685e.png)
